centml/compiler/backend.py

```diff
@@ -76,14 +80,23 @@ def _download_model(self, model_id: str) -> CompiledGraph:
         download_path = get_backend_compiled_forward_path(model_id)
         with open(download_path, "wb") as f:
             f.write(download_response.content)
-            return pickle.loads(download_response.content)
+        return torch.load(download_path)
 
     def _compile_model(self, model_id: str):
-        compile_response = requests.post(
-            url=f"{config_instance.SERVER_URL}/submit/{model_id}",
-            files={"model": pickle.dumps(self.module), "inputs": pickle.dumps(self.inputs)},
-            timeout=config_instance.TIMEOUT,
-        )
+        # The model should have been saved using torch.save when we found the model_id
+        if not os.path.isfile(self.serialized_model_path):
+            raise Exception(f"Model not saved at path {self.serialized_model_path}")
+
+        # Serialize inputs to disk
+        torch.save(self.inputs, self.serialized_input_path, pickle_protocol=config_instance.PICKLE_PROTOCOL)
+
+        with open(self.serialized_model_path, 'rb') as model_file, open(self.serialized_input_path, 'rb') as input_file:
+            compile_response = requests.post(
+                url=f"{config_instance.SERVER_URL}/submit/{model_id}",
+                files={"model": model_file, "inputs": input_file},
```

Comment:
is this gonna work? do you not need to read the files?

Reply:
nope you don't need to, this is valid.

I did this hoping it wouldn't load it into memory, but I think it still does.
regardless, I kept it because it's a bit nicer looking then adding `read()`, but this will probably be changed when I start sending files in chunks